MetaMask/metamask-extension
# Use EIP-1191 checksums

Fixes: https://github.com/rsksmart/rsk-gitcoin-hackathon-2021/issues/11

Explanation:  See [EIP-1191](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1191.md). MetaMask accepts chain ID in network configuration, but does not currently use it in checksum calculation. The PR adds this feature.

![image](https://user-images.githubusercontent.com/44864521/117780426-3d97e700-b25d-11eb-998a-e7e008145902.png)


Manual testing steps:  
  - Add custom network, specify chain ID.
  - Toggle "Checksum uses Chain ID" checkbox and see updated checksum in one of many places where it is rendered (export private key modal, confirm remove account etc.)
